test/MinterDeploy.t.sol

```diff
@@ -0,0 +1,187 @@
+        uint256 deployDate = 1_757_696_400;
+        vm.warp(deployDate); // Sept 12th 2025
+
+        uint256 startTimestamp = vm.getBlockTimestamp();
+        console2.log("startTimestamp", startTimestamp);
+        uint256 mintDeadline = mintAllowedTimestamp + 1 days; // Wednesday, October 15, 2025 1:00:00 PM GMT
+        assertGt(mintDeadline, hypMinter.mintAllowedTimestamp());
+
+        // Pre-requisite. Set the network middleware to be HypMinter via AccessManager
+        TimelockController _network = TimelockController(payable(SYMBIOTIC_NETWORK));
+
+        bytes memory setMiddlewareData = abi.encodeCall(NetworkMiddlewareService.setMiddleware, (address(hypMinter)));
+        bytes memory networkScheduleData = abi.encodeCall(
+            TimelockController.schedule,
+            (address(networkMiddlewareService), 0, setMiddlewareData, bytes32(0), bytes32(0), 0 days)
+        );
+        bytes memory accessManagerScheduleData = abi.encodeCall(
+            AccessManager.schedule, (SYMBIOTIC_NETWORK, networkScheduleData, uint48(vm.getBlockTimestamp()))
+        );
+
+        // Multisig B can call schedule on network via accessManager
+        vm.prank(multisigB);
+        accessManager.schedule(address(_network), networkScheduleData, uint48(vm.getBlockTimestamp()) + 7 days);
+        skip(7 days);
+        vm.prank(multisigB);
+        accessManager.execute(address(_network), networkScheduleData);
+
+        vm.prank(makeAddr("alice"));
+        _network.execute({
+            target: address(networkMiddlewareService),
+            value: 0,
+            payload: setMiddlewareData,
+            predecessor: bytes32(0),
+            salt: bytes32(0)
+        });
+
+        assertEq(networkMiddlewareService.middleware(SYMBIOTIC_NETWORK), address(hypMinter));
+
+        // Give HypMinter the `MINTER_ROLE` on HYPER contract through the AccessManager
+        // 1. Schedule a grantRole operation via the AccessManager
+        vm.warp(startTimestamp); // Do this the same day as the middleware operations
+        vm.prank(multisigB);
+        bytes memory data = abi.encodeCall(AccessControl.grantRole, (keccak256("MINTER_ROLE"), address(hypMinter)));
+        accessManager.schedule(address(HYPER), data, 0);
+
+        // 2. Execute the operation
+        vm.prank(multisigB);
+        skip(30 days);
+        accessManager.execute(address(HYPER), data);
+
+        console2.log("HypMinter gets MINTER_ROLE at: ", vm.getBlockTimestamp());
+        assertGt(mintDeadline, vm.getBlockTimestamp());
+
+        // Fast forward to after the start time
+        vm.warp(Math.max(vm.getBlockTimestamp(), hypMinter.mintAllowedTimestamp() + 1));
+
+        // Three mints
+        deal(address(HYPER), address(hypMinter), MINT_AMOUNT); // We send the already minted amount to the contract
+        uint256 initialBalance = HYPER.balanceOf(address(REWARDS));
+        hypMinter.mint();
+        hypMinter.mint();
+        hypMinter.mint();
+
```

Comment:
can you assert that `mint` reverts after this?

Reply:
Done